app-web/src/pages/topics.js

```diff
@@ -24,9 +24,48 @@ import Main from '../components/Page/Main';
 import withResourceQuery from '../hoc/withResourceQuery';
 import Layout from '../hoc/Layout';
 import { getFirstNonExternalResource } from '../utils/helpers';
+import TableofContents from '../components/TopicPreview/TableofContents';
+import styled from '@emotion/styled';
+import Switch from 'react-switch';
+import { navigate } from 'gatsby';
 
-export const TopicsPage = ({ data }) => {
+const OutsideBorder = styled.div`
+  top-margin: 15px;
+  padding: 7px;
+  border-bottom: 1px solid rgba(0, 0, 0, 0.5);
+`;
+
+const AccordionList = styled.ul`
+  margin: 10px;
+  padding: 10px;
+  background-color: #fff;
+  max-width: 650px;
+  overflow: scroll;
+  border-top: 1px solid rgba(#000, 0.1);
+`;
+
+const ViewMode = styled.span`
+  font-size: 18px;
+  padding: 10px;
+  margin-top: 3px;
+`;
+
+export const TopicsPage = ({ data, location }) => {
   let topics = flattenGatsbyGraphQL(data.allTopicRegistryJson.edges);
+  const options = { cardview: '?v=cards', listview: '?v=list' };
```

Comment:
I like the idea of having a constant for view modes. it would be better to be labeled as a constant such as `const VIEW_MODES = { ...}`